src/commands/staff/sudoreply.ts

```diff
@@ -1,39 +1,106 @@
 import { PVQuestion } from '@lib/types/PVQuestion';
-import { BOT, DB, PREFIX } from '@root/config';
-import { staffPerms } from '@lib/permissions';
-import { Message, MessageEmbed } from 'discord.js';
-import { Command } from '@root/src/lib/types/Command';
+import { BOT, DB, MAINTAINERS } from '@root/config';
+import { ADMIN_PERMS, staffPerms, STAFF_PERMS } from '@lib/permissions';
+import { ApplicationCommandOptionData, ApplicationCommandPermissionData, CommandInteraction, GuildChannel, Message, MessageEmbed, TextChannel, ThreadChannel } from 'discord.js';
+import { Command } from '@lib/types/Command';
+import { Course } from '@lib/types/Course';
 
 export default class extends Command {
 
 	description = `Reply to a question asked through ${BOT.NAME}.`;
 	usage = '<questionID> <response>';
 	extendedHelp = 'Responses get sent to the askers DMs. This command will tell you it failed if it cannot send the DM.';
 	runInDM = false;
-	aliases = ['sreply'];
 
-	permissions(msg: Message): boolean {
-		return staffPerms(msg);
-	}
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'questionid',
+			description: 'ID of question you are replying to',
+			type: 'INTEGER',
+			required: true
+		},
+		{
+			name: 'response',
+			description: 'Response to the question',
+			type: 'STRING',
+			required: true
+		}
+	]
+
+	tempPermissions: ApplicationCommandPermissionData[] = [STAFF_PERMS, ADMIN_PERMS];
+
+	async tempRun(interaction: CommandInteraction): Promise<Message | void> {
+		const question: PVQuestion = await interaction.client.mongo.collection<PVQuestion>(DB.PVQ)
+			.findOne({ questionId: `${interaction.options.getInteger('questionid')}` });
+		const response = interaction.options.getString('response');
+		const bot = interaction.client;
+
+		if (interaction.channel.type !== 'GUILD_TEXT') {
+			return interaction.reply({
+				content: `Something went wrong. Please contact ${MAINTAINERS} for help.`, ephemeral: true
```

Comment:
I feel like a more specific error response works here considering this runs if the interaction is not in a regular message channel.